circuits/circuits/sig.circom

```diff
@@ -0,0 +1,44 @@
+pragma circom 2.1.5;
+
+// TODO: do we need to check that all registers of r,s fit in n bits?
+// Note: The n-bit check is done for pubkey in FlattenPubkey circuit.
+template VerifySignature(n, k) {
+    signal input r[k];
+    signal input s[k];
+    signal input msghash[k];
+    signal input pubkey[2][k];
+
+    signal input leaf_coins[2];
+    signal input receiver;
+    signal input signer;
+
+    // message hash check
+    {
+        signal msghash_computed <== Poseidon(3)(inputs <== [leaf_coins[0], leaf_coins[1], receiver]);
+
+        var cumulativeM = 0;
+        var nPow = 1;
+        for (var i=0; i<k; i++) {
+            _ <== Num2Bits(n)(in <== msghash[i]);
+            cumulativeM += msghash[i]*nPow;
+            nPow *= n;
```

Comment:
```suggestion
            nPow *= 2 ** n;
```
I'm pretty sure this was why your hash was failing. Still doesn't solve the overflow issue that we mentioned last time! Did you implement the fix (register by register comparison) in another PR?

Reply:
this has been done in a nova PRs. and this is a complete fix as the poseidon hash is in the range so its bigint combination also won't overflow.